### ISSUE TITLE
email.utils.getaddresses() rejects email addresses with "," in name

### ISSUE THREAD

**Issue by felixxm** [python/cpython#106669]
# Bug report

`email.utils.getaddresses()` returns `('', '')` for email addresses with `,` in a real name, e.g.
```pycon
>>> from email.utils import getaddresses
>>> getaddresses(('"Sürname, Firstname" <to@example.com>',))
[('', '')]
```

Regression in 18dfbd035775c15533d13a98e56b1d2bf5c65f00.

# Your environment

- CPython versions tested on: 3.12.0b4
- Operating system and architecture: x86_64 GNU/Linux

### Linked PRs
* gh-106733
* gh-106941

**Comment by felixxm** [comment]
A regression test:
```diff
diff --git a/Lib/test/test_email/test_email.py b/Lib/test/test_email/test_email.py
index 5238944d6b..30ade53a9a 100644
--- a/Lib/test/test_email/test_email.py
+++ b/Lib/test/test_email/test_email.py
@@ -3319,6 +3319,22 @@ def test_getaddresses(self):
            [('Al Person', 'aperson@dom.ain'),
             ('Bud Person', 'bperson@dom.ain')])
 
+    def test_getaddresses_comma_in_name(self):
+        self.assertEqual(
+            utils.getaddresses(
+                [
+                    '"Bud, Person" <bperson@dom.ain>',
+                    'aperson@dom.ain (Al Person)',
+                    '"Mariusz Felisiak" <to@example.com>',
+                ]
+            ),
+            [
+                ('Bud Person', 'bperson@dom.ain'),
+                ('Al Person', 'aperson@dom.ain'),
+                ('Mariusz Felisiak', 'to@example.com'),
+            ],
+        )
+
     def test_getaddresses_parsing_errors(self):
         """Test for parsing errors from CVE-2023-27043"""
         eq = self.assertEqual
```

**Comment by rouilj** [comment]
Thanks for finding this. It blew up the regression testing for the Roundup Issue Tracker during my release yesterday. Specifically for:

>>> email.utils.getaddresses(['"Bork, Chef" <chef@bork.bork.bork>'])
[('', '')]

Any idea if there will be a beta-5 to fix this? If not I could use some ideas on how to
handle/decorate the three tests that are failing when run on 3.12.

Thanks.

**Comment by gpshead** [comment]
If you add a test skip decorator, I'd do it specifically for 3.12beta4. Something like this:

```python
@unittest.skipIf(sys.version_info == (3, 12, 0, 'beta', 4), "https://github.com/python/cpython/issues/106669")
def test_oops(self): ...
```

It does not look like there is an easy acceptable workaround for the bug as the source of the problem appears to be the comma counting logic added near the end of `email.utils.getaddresses()`.

It's up to @Yhg1s to decide if this warrants a beta5 or not. I've prepared a rollback of the change that caused it.

**Comment by michel-slm** [comment]
This also breaks b4 very similarly to Django; mentioning here for reference

https://bugzilla.redhat.com/show_bug.cgi?id=2226159

```
    def test_header_wrapping(sampledir, hval, verify, tr):
        hname = 'To' if '@' in hval else "X-Header"
        wrapped = b4.LoreMessage.wrap_header((hname, hval), transform=tr)
>       assert wrapped.decode() == f'{hname}: {verify}'
E       assert 'To: ' == 'To: foo@exam...@example.com>'
E         + To: 
E         - To: foo@example.com, Foo Bar <bar@example.com>, 
E         -  =?utf-8?q?F=C3=B4o_Baz?= <baz@example.com>, "Quux, Foo" <quux@example.com>
tests/test___init__.py:171: AssertionError
```

**Comment by gpshead** [comment]
The rollbacks were merged and will appear in 3.12.0rc1.
